auditing/auditing-interceptor.go

```diff
@@ -0,0 +1,210 @@
+package auditing
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/emicklei/go-restful/v3"
+	"github.com/google/uuid"
+	"github.com/metal-stack/metal-lib/rest"
+	"github.com/metal-stack/security"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+const (
+	Exclude string = "exclude-from-auditing"
+)
+
+func UnaryServerInterceptor(a Auditing, logger *zap.SugaredLogger, shouldAudit func(fullMethod string) bool) grpc.UnaryServerInterceptor {
```

Comment:
Should these grpc interceptors be placed in metal-stack-cloud/api-server instead? This would remove some dependencies from the metal-lib.